File: 4.8-bc/py/stage-3-hexagonal/app/catalog/app/service.py
"""Catalog use cases — List products."""
from __future__ import annotations

from app.catalog.domain.product import Product
from app.catalog.domain.repository import Repository


class Service:
    def __init__(self, repo: Repository) -> None:
        self._repo = repo

    async def list(self) -> list[Product]:
        return await self._repo.list()

File: 4.8-bc/go/stage-2-feature-first/app/catalog/service.go
// Package catalog — Catalog feature: продукти, категорії.
package catalog

import "context"

type Service struct {
	repo *Repository
}

func NewService(repo *Repository) *Service {
	return &Service{repo: repo}
}

func (s *Service) List(ctx context.Context) ([]Product, error) {
	return s.repo.List(ctx)
}
